File: src/main/java/catchla/yep/preference/AccountInfoPreference.java
package catchla.yep.preference;

import android.accounts.Account;
import android.content.Context;
import android.content.Intent;
import android.support.v7.preference.Preference;
import android.support.v7.preference.PreferenceViewHolder;
import android.util.AttributeSet;
import android.widget.ImageView;
import android.widget.TextView;

import javax.inject.Inject;

import catchla.yep.Constants;
import catchla.yep.R;
import catchla.yep.activity.ProfileEditorActivity;
import catchla.yep.model.User;
import catchla.yep.util.ImageLoaderWrapper;
import catchla.yep.util.Utils;
import catchla.yep.util.dagger.GeneralComponentHelper;

/**
 * Created by mariotaku on 15/5/11.
 */
public class AccountInfoPreference extends Preference implements Constants {

    private final Account mAccount;
    private final User mAccountUser;
    @Inject
    ImageLoaderWrapper mImageLoader;

    public AccountInfoPreference(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        GeneralComponentHelper.build(context).inject(this);
        setLayoutResource(R.layout.layout_preference_account_info);
        mAccount = Utils.getCurrentAccount(context);
        mAccountUser = Utils.getCurrentAccountUser(context);
    }

    public AccountInfoPreference(Context context, AttributeSet attrs) {
        this(context, attrs, android.R.attr.preferenceStyle);
    }

    public AccountInfoPreference(Context context) {
        this(context, null);
    }

    @Override
    public void onBindViewHolder(final PreferenceViewHolder holder) {
        super.onBindViewHolder(holder);
        if (mAccount != null) {
            final ImageView profileImageView = (ImageView) holder.findViewById(R.id.account_profile_image);
            final TextView nameView = (TextView) holder.findViewById(R.id.account_name);
            final TextView introductionView = (TextView) holder.findViewById(R.id.account_introduction);
            mImageLoader.displayProfileImage(mAccountUser.getAvatarThumbUrl(), profileImageView);
            nameView.setText(mAccountUser.getNickname());
            introductionView.setText(mAccountUser.getIntroduction());
        }
    }

    @Override
    protected void onClick() {
        final Context context = getContext();
        final Intent intent = new Intent(context, ProfileEditorActivity.class);
        intent.putExtra(EXTRA_ACCOUNT, mAccount);
        context.startActivity(intent);
    }
}

File: src/main/kotlin/catchla/yep/preference/AccountInfoPreference.kt
package catchla.yep.preference

import android.accounts.Account
import android.content.Context
import android.content.Intent
import android.support.v7.preference.Preference
import android.support.v7.preference.PreferenceViewHolder
import android.util.AttributeSet
import android.widget.ImageView
import android.widget.TextView
import catchla.yep.Constants
import catchla.yep.R
import catchla.yep.activity.ProfileEditorActivity
import catchla.yep.model.User
import catchla.yep.util.ImageLoaderWrapper
import catchla.yep.util.Utils
import catchla.yep.util.dagger.GeneralComponentHelper
import javax.inject.Inject

/**
 * Created by mariotaku on 15/5/11.
 */
class AccountInfoPreference @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = android.R.attr.preferenceStyle) :
        Preference(context, attrs, defStyleAttr), Constants {

    private val mAccount: Account?
    private val mAccountUser: User?
    @Inject
    lateinit internal var mImageLoader: ImageLoaderWrapper

    init {
        GeneralComponentHelper.build(context).inject(this)
        layoutResource = R.layout.layout_preference_account_info
        mAccount = Utils.getCurrentAccount(context)
        mAccountUser = Utils.getCurrentAccountUser(context)
    }

    override fun onBindViewHolder(holder: PreferenceViewHolder) {
        super.onBindViewHolder(holder)
        if (mAccount != null && mAccountUser != null) {
            val profileImageView = holder.findViewById(R.id.account_profile_image) as ImageView
            val nameView = holder.findViewById(R.id.account_name) as TextView
            val introductionView = holder.findViewById(R.id.account_introduction) as TextView
            mImageLoader.displayProfileImage(mAccountUser.avatarThumbUrl, profileImageView)
            nameView.text = mAccountUser.nickname
            introductionView.text = mAccountUser.introduction
        }
    }

    override fun onClick() {
        val context = context
        val intent = Intent(context, ProfileEditorActivity::class.java)
        intent.putExtra(Constants.EXTRA_ACCOUNT, mAccount)
        context.startActivity(intent)
    }
}
